src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -1465,13 +1417,74 @@ private void parseItemset(QuestionDef q, Element e) {
+        String labelXpath = child.getAttributeValue("", REF_ATTR);
+        boolean labelItext = false;
+
+        if (XFormUtils.showUnusedAttributeWarning(child, labelUA)) {
+            reporter.warning(XFormParserReporter.TYPE_UNKNOWN_MARKUP, XFormUtils.unusedAttWarning(child, labelUA), getVagueLocation(child));
+        }
+
+        if (labelXpath != null) {
+            if (labelXpath.startsWith("jr:itext(") && labelXpath.endsWith(")")) {
+                labelXpath = labelXpath.substring("jr:itext(".length(), labelXpath.indexOf(")"));
+                labelItext = true;
+            }
+        } else {
+            throw new XFormParseException("<label> in <itemset> requires 'ref'");
+        }
+
+        XPathPathExpr labelPath = XPathReference.getPathExpr(labelXpath);
+        itemset.labelRef = FormInstance.unpackReference(getAbsRef(new XPathReference(labelPath), itemset.nodesetRef));
+        itemset.labelExpr = new XPathConditional(labelPath);
+        itemset.labelIsItext = labelItext;
+    }
+
+    private void parseItemsetCopyElement(Element child, ItemsetBinding itemset, Vector<String> copyUA) {
+        String copyRef = child.getAttributeValue("", REF_ATTR);
+        if (XFormUtils.showUnusedAttributeWarning(child, copyUA)) {
+            reporter.warning(XFormParserReporter.TYPE_UNKNOWN_MARKUP, XFormUtils.unusedAttWarning(child, copyUA), getVagueLocation(child));
+        }
+        if (copyRef == null) {
+            throw new XFormParseException("<copy> in <itemset> requires 'ref'");
+        }
+        itemset.copyRef = FormInstance.unpackReference(getAbsRef(new XPathReference(copyRef), itemset.nodesetRef));
+        itemset.copyMode = true;
+    }
+
+    private void parseItemsetValueElement(Element child, ItemsetBinding itemset, Vector<String> valueUA) {
+        String valueXpath = child.getAttributeValue("", REF_ATTR);
+
+        if (XFormUtils.showUnusedAttributeWarning(child, valueUA)) {
+            reporter.warning(XFormParserReporter.TYPE_UNKNOWN_MARKUP, XFormUtils.unusedAttWarning(child, valueUA), getVagueLocation(child));
+        }
+        if (valueXpath == null) {
+            throw new XFormParseException("<value> in <itemset> requires 'ref'");
+        }
+
+        XPathPathExpr valuePath = XPathReference.getPathExpr(valueXpath);
+        itemset.valueRef = FormInstance.unpackReference(getAbsRef(new XPathReference(valuePath), itemset.nodesetRef));
+        itemset.valueExpr = new XPathConditional(valuePath);
+        itemset.copyMode = false;
+    }
+
+    private void parseItemsetSortElement(Element child, ItemsetBinding itemset) {
+        String sortXpathString = child.getAttributeValue("", REF_ATTR);
+        if (sortXpathString == null) {
+            throw new XFormParseException("<sort> in <itemset> requires 'ref'");
+        }
+
+        XPathPathExpr sortPath = XPathReference.getPathExpr(sortXpathString);

```

Comment:
Hey yea I think I'm going to leave this for now, for a couple of reasons. Mostly I just don't think I'm going to have time to do this before tomorrow without displacing other work that needs to get finished. But also from the perspective of the HQ configuration for the user, if this is going to be a 3rd box below those for the "Value Field" and "Display Text Field", it seems confusing to make it so that the way in which the users specifies it is actually not the same. I think there are ways to resolve that if more thought was to go into the HQ UI, but maybe to be decided at a later date if it becomes needed.